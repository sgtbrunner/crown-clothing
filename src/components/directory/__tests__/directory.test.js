import React from 'react';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';

import Directory from '../directory.component';

const mockStore = configureMockStore();

const sections = [
  {
    title: 'hats',
    imageUrl: 'https://i.ibb.co/cvpntL1/hats.png',
    id: 1,
    linkUrl: 'hats',
  },
  {
    title: 'jackets',
    imageUrl: 'https://i.ibb.co/px2tCc3/jackets.png',
    id: 2,
    linkUrl: 'jackets',
  },
  {
    title: 'shoes',
    imageUrl: 'https://i.ibb.co/0jqHpnp/sneakers.png',
    id: 3,
    linkUrl: 'shoes',
  },
  {
    title: 'womens',
    imageUrl: 'https://i.ibb.co/GCCdy8t/womens.png',
    size: 'large',
    id: 4,
    linkUrl: 'womens',
  },
  {
    title: 'mens',
    imageUrl: 'https://i.ibb.co/R70vBrQ/men.png',
    size: 'large',
    id: 5,
    linkUrl: 'mens',
  },
];

const store = mockStore({
  directory: {
    sections,
  },
});

const mountComponent = () =>
  render(
    <Provider store={store}>
      <Router basename="/">
        <Directory />
      </Router>
    </Provider>
  );

describe('Directory component', () => {
  it('should match snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render five (5) menu items', () => {
    mountComponent();

    const menuItems = screen.getAllByTestId('menu-item');
    expect(menuItems).toHaveLength(5);
  });
});
