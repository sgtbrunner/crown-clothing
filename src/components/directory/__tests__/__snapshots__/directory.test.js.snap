// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory component should match snapshot 1`] = `
<div>
  <div
    class="sc-jSFjdj eiSdVN"
  >
    <div
      class="sc-bdnxRM NchVL"
      data-testid="menu-item"
    >
      <div
        class="sc-gtsrHT gSOdEr background-image"
      />
      <div
        class="sc-dlnjwi jgWYvt"
      >
        <span
          class="sc-hKFxyN heyTLl"
        >
          HATS
        </span>
        <span
          class="sc-eCApnc llITcv"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      class="sc-bdnxRM NchVL"
      data-testid="menu-item"
    >
      <div
        class="sc-gtsrHT gHihWz background-image"
      />
      <div
        class="sc-dlnjwi jgWYvt"
      >
        <span
          class="sc-hKFxyN heyTLl"
        >
          JACKETS
        </span>
        <span
          class="sc-eCApnc llITcv"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      class="sc-bdnxRM NchVL"
      data-testid="menu-item"
    >
      <div
        class="sc-gtsrHT gEMxoM background-image"
      />
      <div
        class="sc-dlnjwi jgWYvt"
      >
        <span
          class="sc-hKFxyN heyTLl"
        >
          SHOES
        </span>
        <span
          class="sc-eCApnc llITcv"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      class="sc-bdnxRM dDDrsu"
      data-testid="menu-item"
    >
      <div
        class="sc-gtsrHT khwRlI background-image"
      />
      <div
        class="sc-dlnjwi jgWYvt"
      >
        <span
          class="sc-hKFxyN heyTLl"
        >
          WOMENS
        </span>
        <span
          class="sc-eCApnc llITcv"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      class="sc-bdnxRM dDDrsu"
      data-testid="menu-item"
    >
      <div
        class="sc-gtsrHT gvFqFL background-image"
      />
      <div
        class="sc-dlnjwi jgWYvt"
      >
        <span
          class="sc-hKFxyN heyTLl"
        >
          MENS
        </span>
        <span
          class="sc-eCApnc llITcv"
        >
          SHOP NOW
        </span>
      </div>
    </div>
  </div>
</div>
`;
